src/LinearOperators.jl

```diff
@@ -339,6 +339,49 @@ function opDiagonal(nrow :: Int, ncol :: Int, d :: Vector)
   return D
 end
 
+
+import Base.hcat
+import Base.vcat
+function hcat(A::LinearOperator,B::LinearOperator)
+	if A.nrow != B.nrow
+		error("numbers of rows must be the same")
+	end
+	if A.dtype != B.dtype
+		error("data type must be the same")
+	end
```

Comment:
Other operations with operators use `promote_type()` to infer the smallest common type that can represent all operands. It would be best to do the same here.